lib/adapters/definition-adapter.ts

```diff
@@ -68,11 +68,15 @@ export default class DefinitionAdapter {
    * @param locationResult Either a single {Location} object or an {Array} of {Locations}.
    * @returns An {Array} of {Location}s or {null} if the locationResult was null.
    */
-  public static normalizeLocations(locationResult: Location | Location[]): Location[] | null {
+  public static normalizeLocations(
+    locationResult: Location | Location[] | LocationLink[] | null
+  ): Location[] | LocationLink[] | null {
     if (locationResult == null) {
       return null
     }
-    return (Array.isArray(locationResult) ? locationResult : [locationResult]).filter((d) => d.range.start != null)
+    return (Array.isArray(locationResult) ? (locationResult as any[]) : [locationResult]).filter(
```

Comment:
This cast to `any` is very suspicious. Casting shouldn't be used in normal cases.

Reply:
Instead of not using the casting, I added a type guard function.